frontend/src/reports/query/Query.svelte

```diff
@@ -18,16 +18,20 @@
   const results: Record<string, Result<QueryResult, string>> = {};
   const is_open: Record<string, boolean> = {};
 
-  onMount(() =>
+  onMount(() => {
     searchParams.subscribe((s) => {
       const search_query_string = s.get("query_string") ?? "";
       // Set the query string to the value from the URL query if that changes (e.g. on navigation).
       if (search_query_string !== query_string) {
         query_string = search_query_string;
         submit();
       }
-    }),
-  );
+    });
+
+    filter_params.subscribe(() => {
+      submit();
```

Comment:
we also need to handle the results cache and the other results that are displayed

Reply:
Sorry, I don't follow. Can you elaborate?

Reply:
When you run multiple queries in sequence, all those results are kept around (in the object `results`). Those results also need to be updated / invalidated when the filter changes

Reply:
Before Fava 1.29, when I update the filter only the latest result is updated. Can you confirm the intended behaviour is to re-run all history queries?

Reply:
Re-running all of them is probably a bit excessive. I'll change this to re-run all for which the result is currently opened.